### PR TITLE
Connector with modifier options

### PR DESCRIPTION
**Pull request by bre1470**
- Moved async modifier attachment from pool to connectors.
- Updated climate demo.
- Updated doclets.
- Removed API duplicates

### CODE REVIEW

File: samples/dashboards/demo/climate/demo.js [hunk 1]
```diff
@@ -977,20 +976,29 @@ async function updateBoard(board, city, column, scale, newData) {
...
     const timeRangeMax = timeRangeSelector.chart.axes[0].max;
     const timeRangeMin = timeRangeSelector.chart.axes[0].min;
-    await selectionTable.setModifier(new RangeModifier({
-        modifier: 'Range',
-        ranges: [{
+    const selectionModifier = selectionTable.getModifier();
+
+    if (
+        !selectionModifier ||
+        selectionModifier.options.ranges[0].maxValue !== timeRangeMax ||
```
Comment (sebastianbochan) [inline samples/dashboards/demo/climate/demo.js:991]: Shouldn't we check that `selectionModifier.options` exist?

**Review by sebastianbochan — APPROVED** [review]

File: samples/dashboards/demo/climate/demo.js (hunk 1 above)
Comment (bre1470) [reply samples/dashboards/demo/climate/demo.js:991]: `DataModifier.options` always exists. `ranges` is a default. The only thing that could fail is accessing index 0, but we know that we have set it.

File: samples/dashboards/demo/climate/demo.js [hunk 2]
```diff
@@ -977,20 +976,29 @@ async function updateBoard(board, city, column, scale, newData) {
...
-    selectionTable.setColumns(cityTable.modified.getColumns(), 0);
     const timeRangeMax = timeRangeSelector.chart.axes[0].max;
     const timeRangeMin = timeRangeSelector.chart.axes[0].min;
-    await selectionTable.setModifier(new RangeModifier({
-        modifier: 'Range',
-        ranges: [{
+    const selectionModifier = selectionTable.getModifier();
+
+    if (
+        !selectionModifier ||
```
Comment (karolkolodziej) [inline samples/dashboards/demo/climate/demo.js:990]: image
I get that so i presume the condition `!selectionModifier ||` isn't ideal

**Review by karolkolodziej — CHANGES_REQUESTED** [review]

**Review by pawellysy — APPROVED** [review]
Looks good! 👍

File: samples/dashboards/demo/climate/demo.js [hunk 3]
```diff
@@ -965,6 +969,7 @@ async function updateBoard(board, city, column, scale, newData) {
         selectionGrid,
         cityChart
     ] = board.mountedComponents.map(c => c.component);
+    console.log(await dataPool.getConnector('Range Selection'));
```
Comment (karolkolodziej) [inline samples/dashboards/demo/climate/demo.js:972]: do we need that?

**Review by karolkolodziej — APPROVED** [review]

File: samples/dashboards/demo/climate/demo.js (hunk 3 above)
Comment (bre1470) [reply samples/dashboards/demo/climate/demo.js:972]: No 🙂

**Pull request merged**